sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -18,6 +18,288 @@
 void enableZeTracing();
 void disableZeTracing();
 
+// TODO: Move these functions to the UR plugin once it's updated to include the
+// changes from oneapi-src/unified-runtime#849. urEnqueueCooperativeKernelLaunch
+// llvm/sycl/plugins/unified_runtime/ur/adapters/level_zero/kernel.cpp
+namespace pi2ur {
+pi_result piextKernelSuggestMaxCooperativeGroupCount(pi_kernel Kernel,
+                                                     pi_uint32 *GroupCountRet) {
+  ur_kernel_handle_t UrKernel = reinterpret_cast<ur_kernel_handle_t>(Kernel);
+  const auto Result = zeKernelSuggestMaxCooperativeGroupCount(
+      UrKernel->ZeKernel, GroupCountRet);
+  return ur2piResult(ze2urResult(Result));
+}
+
+// TODO: This function is a modified version of urEnqueueKernelLaunch in
+// llvm/sycl/plugins/unified_runtime/ur/adapters/level_zero/kernel.cpp. The two
```

Comment:
can the refactor be done on this PR or several small patches? Like one patch would extract some common code creating new helpers, laying out the code so then urEnqueueCooperativeKernelLaunch and urEnqueueKernelLaunch can use both those helpers.